classroom/snippets/src/main/java/ModifyAttachmentsStudentSubmission.java

```diff
@@ -0,0 +1,93 @@
+// limitations under the License.
+
+
+// [START classroom_modify_attachments_student_submissions_class]
+
+import com.google.api.client.googleapis.json.GoogleJsonError;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.api.services.classroom.Classroom;
+import com.google.api.services.classroom.ClassroomScopes;
+import com.google.api.services.classroom.model.Attachment;
+import com.google.api.services.classroom.model.Link;
+import com.google.api.services.classroom.model.ModifyAttachmentsRequest;
+import com.google.api.services.classroom.model.StudentSubmission;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+
+/* Class to demonstrate the use of Classroom ModifyAttachments StudentSubmissions API. */
+public class ModifyAttachmentsStudentSubmission {
+  /**
+   * Lists student submissions based on courseId, courseWorkId, and userId.
+   *
+   * @param courseId - identifier of the course.
+   * @param courseWorkId - identifier of the course work.
+   * @param id - identifier of the student submission.
+   * @return - the modified student submissions.
+   * @throws IOException - if credentials file not found.
+   */
+  public static StudentSubmission modifyAttachments(String courseId, String courseWorkId, String id)
+      throws IOException {
+    /* Load pre-authorized user credentials from the environment.
+     TODO(developer) - See https://developers.google.com/identity for
+      guides on implementing OAuth2 for your application. */
+    GoogleCredentials credentials = GoogleCredentials.getApplicationDefault()
+        .createScoped(Collections.singleton(ClassroomScopes.CLASSROOM_COURSEWORK_STUDENTS));
+    HttpRequestInitializer requestInitializer = new HttpCredentialsAdapter(
+        credentials);
+
+    // Create the classroom API client.
+    Classroom service = new Classroom.Builder(new NetHttpTransport(),
+        GsonFactory.getDefaultInstance(),
+        requestInitializer)
+        .setApplicationName("Classroom samples")
+        .build();
+
+    // [START classroom_modify_attachments_student_submissions_code_snippet]
+
+    StudentSubmission studentSubmission = null;
+    try {
+      // Create ModifyAttachmentRequest object that includes a new attachment with a link.
+      Link link = new Link().setUrl("https://en.wikipedia.org/wiki/Irrational_number");
+      Attachment attachment = new Attachment().setLink(link);
+      ModifyAttachmentsRequest modifyAttachmentsRequest = new ModifyAttachmentsRequest()
+          .setAddAttachments(Arrays.asList(attachment));
+      studentSubmission = service.courses().courseWork().studentSubmissions().modifyAttachments(
```

Comment:
Can we add a comment somewhere explaining why a StudentSubmission object is returned? I presume it's meant to contain the new modification as a confirmation, but would be nice for the reader to know what to expect.